addons/suggest_locations/models/suggest_by_product.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+from odoo import models, fields, _
+from itertools import groupby
+from .suggest_locations_policy import SuggestLocationPolicy, get_selection
+
+
+class ByProduct(SuggestLocationPolicy):
+    """Find locations based on other placed that product is stored"""
+
+    @classmethod
+    def name(cls):
+        return "by_product"
+
+    def get_values_from_mls(self, mls):
+        return {
+            "product": mls.product_id.ensure_one(),
+            "parent_location": mls.u_picking_type_id.default_location_dest_id.ensure_one(),
+        }
+
+    def _get_product_from_dict(self, values):
+        """Get the product from the values"""
+        product_id = values.get("product_id")
+        if not product_id:
+            raise ValueError(_("No product found"))
+
+        if isinstance(product_id, int):
+            return self.env["product.product"].browse(product_id)
+        elif isinstance(product_id, models.BaseModel):
+            return product_id.ensure_one()
+
+    def _get_picking_from_dict(self, values):
+        """Get the picking from the values 
+        Note: Cannot just use the picking because in _prepare_move_lines we 
+        call suggest_locations but couldn't add a field picking_rec, already
+        has picking_id
+        """
+        picking_id = values.get("picking_id")
+        if not picking_id:
+            raise ValueError(_("No picking found"))
+        if isinstance(picking_id, int):
+            picking = self.env["stock.picking"].browse(picking_id)
+        elif isinstance(picking_id, models.BaseModel):
+            picking = picking_id
+        return picking
+
+    def get_values_from_dict(self, values):
+        picking = self._get_picking_from_dict(values)
+        return {
+            "product": self._get_product_from_dict(values),
+            "parent_location": picking.picking_type_id.default_location_dest_id.ensure_one(),
```

Comment:
In this case, picking should be just one, so we don't need ensure_one.

Reply:
Also, shouldn't parent_location be the picking.location_dest_id ? the one in the picking and the one in the picking_type do not necessarily match